nomad/state/events.go

```diff
@@ -56,12 +56,15 @@ func eventFromChange(change memdb.Change) (structs.Event, bool) {
 			if !ok {
 				return structs.Event{}, false
 			}
+
+			// Copy token and empty out secret ID
```

Comment:
If we always want to copy and empty the token before passing it into `NewACLTokenEvent`, maybe it'd be worth moving that into that constructor and have `NewACLTokenEvent` return both the `ACLTokenEvent` and a new token that's been copied and emptied? That keeps all the security-sensitive steps together in one call.

Reply:
Good idea, we don't actually need the new token returned, will update to do it all in `NewACLTokenEvent`